hw2/03-count.js

```diff
@@ -1,3 +1,37 @@
 // Add your code here
 
-input.addEventListener('keydown', handleKeyDown);
+const highlightClass = "bg-warning";
+
+const textContainer = document.getElementById("text-container");
+// Restructure the original text container
+const originalTextSpanList = textContainer.innerText.split(" ").map((word) => {
+  const newSpan = document.createElement("span");
+  newSpan.innerText = word;
+  newSpan.style.transitionDuration = "100ms";
+  return newSpan;
+});
```

Comment:
curious but are you thinking of dynamically add/remove `span` for matching words ❓ 
if yes, I think that will kill performance because we are modifying the DOM directly and every single time the user types. 
instead, adding/removing className would be much more performance in my opinion

maybe I miss something ❓

Reply:
of course in a larger-scale solution, adding extra DOM elements is not ideal at all because it will mess with the intentional layout, but... that's another story for another day 😄

Reply:
It just felt off to see so many span elements in a single parent div, one per word. The screenshot below is what I was looking at when I commented this. 

![spans](https://github.com/ptmdmusique/cs564-frontend-homework/assets/38127011/02e591fb-1775-496f-9ff2-abc21b9dae22)